### ISSUE TITLE
Strange behaviour of strong charges in GB-Neck2 implicit solvent

### ISSUE THREAD

**Issue by tristanic** [openmm/openmm#2775]
I'm attaching a serialised system of two phosphatidic acids parameterised using GAFF2 (ANTECHAMBER via phenix.elbow, for the record). I realise part of the problem is that they're non-physiological (at near-neutral pH the phosphates should be protonated with an overall -1 charge; this has them at -2), but it illustrates a seeming pathology of the implicit solvent environment that I've noticed a few times: strong charges tend to attract *much* more strongly than they should (and sometimes in seemingly paradoxical ways). Here the phosphates stick to each other like glue, requiring substantial force to pull them apart and pulling straight back together from as far as 12A (getting really aggressive from about 8A separation between phosphorous atoms). Charges seem reasonable, and the atoms are all just standard GAFF types:

```
    <Residue name="MC_3PH">
      <Atom charge="-0.909833" name="O13" type="o"/>
      <Atom charge="1.1779" name="P" type="p5"/>
      <Atom charge="-0.909833" name="O14" type="o"/>
      <Atom charge="-0.909833" name="O12" type="o"/>
      <Atom charge="-0.5602" name="O11" type="os"/>
      <Atom charge="0.2054" name="C1" type="c3"/>
```
Any idea what might be going on?

[system.xml.zip](https://github.com/openmm/openmm/files/4891681/system.xml.zip)

**Comment by tristanic** [comment]
Playing around with a magnesium ion near a phosphate, it looks like the positive charge on the phosphate is being screened at a much slower rate than the negative charges on the oxygen. If I put a +1 ion (with the L-J parameters of a magnesium) close to one of the oxygens it shoots away - even if I apply a weak distance restraint with a target of 1.5A and a spring constant of 5 kJ/mol/A**2 the equilibrium distance is about 6.4A. But if I carefully bring it in to VDW contact with *two* oxygens then it's (meta) stable. I also see similar behaviour with the same ion interacting with sulfate. Also, the reason I have the metal ion charge set to +1 is that +2 charges led to ridiculously-strong interactions with just about everything... now I'm wondering if there's a bug leading to mismatched treatment of positive vs. negative charges?

**Comment by proteneer** [comment]
Do you see this behavior if you turn off GB entirely?

**Comment by tristanic** [comment]
No. If I turn off GB it all behaves pretty much exactly as expected. I'm
trying to dig out where the screening parameters originally come from...
the original GB-neck2 paper
(https://pubs.acs.org/doi/full/10.1021/ct3010485) only covers C, O, H
and N, but AMBER has the -0.703 for S and 0.5 for P that OpenMM uses for
non-nucleic acid groups. OpenMM adds nucleic acid-specific values of
1.545 for P and 0.05 for S - I don't know where these come from, but
RNA/DNA do seem to behave much more like what I'd expect.

On 2020-07-08 19:54, Yutong Zhao wrote:

> Do you see this behavior if you turn off GB entirely? 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].
 

Links:
------
[1] https://github.com/openmm/openmm/issues/2775#issuecomment-655695362
[2]

**Comment by tristanic** [comment]
`GBSAGBnForce` is quite well behaved for the same system (plus or minus a magnesium) - phosphates repel each other as expected, magnesium is attracted as expected.

**Comment by tristanic** [comment]
This also appears to explain the strange behaviour a colleague showed me some time back, of a ligand containing both a trifluoromethyl and a sulfonamide group in an MDFF simulation. The trifluoromethyl was being pushed _really_ aggressively out of density even with very strong map forces. Returning to this after having replaced GB-Neck2 with GB-Neck and the same system behaves perfectly. It seems there's something very wrong with the GB-Neck2 pararameterisation for ligand atoms.

**Comment by tristanic** [comment]
Looks like *part* of this problem appears to stem from my failing to do `nbforce.setReactionFieldDielectric(1.0)` when setting up the GBSA force. But the effects of that in GB-Neck are relatively subtle - I don't think that's the full story.

**Comment by peastman** [comment]
Did you ever track this down?  Is it still an open issue?

**Comment by tristanic** [comment]
I'm afraid this seemingly-unending lockdown has put the kibosh on almost all my development activities (poring through code with two bored kids in the house is a skill I'll probably never master). I've been focused almost entirely on applied work since last March. Beyond confirming that the GB-neck implementation seems to work fine while neck2 is problematic, I haven't had a chance to dig further.
________________________________

**Comment by csimmerling** [comment]
just happened to run across this, if anyone revisits it I would be happy to help out. I have some suggestions tha tmight fix the problem, we've looked into this in Amber as well.
